lib/apartmentex/tenant_actions.ex

```diff
@@ -21,6 +33,33 @@ defmodule Apartmentex.TenantActions do
     end
   end
 
+  defp migrate_and_return_status(repo, tenant, direction, opts) do
+    prefix = build_prefix(tenant)
+
+    {status, versions} = handle_database_exceptions fn ->
+      opts_with_prefix = Keyword.put(opts, :prefix, prefix)
+      Ecto.Migrator.run(
+        repo,
+        tenant_migration_folder(repo),
+        direction,
+        opts_with_prefix
+      )
+    end
+
+    {status, prefix, versions}
+  end
+
+  defp handle_database_exceptions(fun) do
+    try do
+      {:ok, fun.()}
+    rescue
+      e in Postgrex.Error ->
+        {:error, Postgrex.Error.message(e)}
+      e in Mariaex.Error ->
+        {:error, Mariaex.Error.message(e)}
```

Comment:
I'm pretty sure this works, but we are not testing MySQL in an integration test. I looked into adding a test for MySQL, but it's harder because I don't think you can create a database in a transactional way like you can with postgres schemas